lib/Controller/SettingsController.php

```diff
@@ -86,4 +95,81 @@ public function setSIPSettings(
 
 		return new DataResponse(null);
 	}
+
+	/**
+	 * Get writable calendars and the default calendar
+	 *
+	 * Required capability: `schedule-meeting`
+	 *
+	 * @return DataResponse<Http::STATUS_OK, array{defaultCalendarUri: ?string, calendars: list<TalkCalendar>}, array{}>
+	 *
+	 * 200: Get a list of calendars
+	 */
+	#[NoAdminRequired]
+	public function getPersonalCalendars(): DataResponse {
+		$calendars = $this->calendarManager->getCalendarsForPrincipal('principals/users/' . $this->userId);
+
+		$defaultCalendarUri = $selectedCalendarUri = null;
+		$selectedCalendar = $this->getSchedulingCalendarFromPropertiesTable($this->userId);
+		if ($selectedCalendar !== false) {
+			$parts = explode('/', rtrim($selectedCalendar, '/'));
+			if (count($parts) === 3 && $parts[0] === 'calendars' && $parts[1] === $this->userId) {
+				$selectedCalendarUri = $parts[2];
+			}
+		}
+
+		$hasPersonal = false;
+		$list = [];
+		foreach ($calendars as $calendar) {
+			if ($calendar->isDeleted()) {
+				continue;
+			}
+
+			if (!($calendar->getPermissions() & Constants::PERMISSION_CREATE)) {
+				continue;
+			}
+
+			if ($calendar->getUri() === 'personal') {
+				$hasPersonal = true;
+			}
+
+			if ($selectedCalendarUri === $calendar->getUri()) {
+				$defaultCalendarUri = $selectedCalendarUri;
+			}
+
+			$list[] = [
+				'uri' => $calendar->getUri(),
+				'name' => $calendar->getDisplayName() ?? $calendar->getUri(),
+				'color' => $calendar->getDisplayColor(),
+			];
+		}
+
+		return new DataResponse([
+			'defaultCalendarUri' => $defaultCalendarUri ?? ($hasPersonal ? 'personal' : null),
+			'calendars' => $list,
+		]);
+	}
+
+	/**
+	 * @param string $userId
+	 * @return false|string
+	 */
+	protected function getSchedulingCalendarFromPropertiesTable(string $userId) {
+		$propertyPath = 'principals/users/' . $userId;
+		$propertyName = '{urn:ietf:params:xml:ns:caldav}schedule-default-calendar-URL';
+
+		$query = $this->db->getQueryBuilder();
+		$query->select('propertyvalue')
+			->from('properties')
+			->where($query->expr()->eq('userid', $query->createNamedParameter($userId)))
+			->andWhere($query->expr()->eq('propertypath', $query->createNamedParameter($propertyPath)))
+			->andWhere($query->expr()->eq('propertyname', $query->createNamedParameter($propertyName)))
+			->setMaxResults(1);
+
+		$result = $query->executeQuery();
+		$property = $result->fetchOne();
+		$result->closeCursor();
+
+		return $property;
```

Comment:
You should go through the DAV app to get the default calendar URI, or you can use the user config:


`$uri = $this->appConfig->getUserValue($userId, 'dav', 'defaultCalendar', CalDavBackend::PERSONAL_CALENDAR_URI);`

Reply:
right, should use that preference,
but going through the dav app is to risky as it's all private api

Reply:
So I toggled this:
![grafik](https://github.com/user-attachments/assets/915dddf7-8020-4f25-b18a-c9c596551497)

But there is no value in oc_preferences

Reply:
I also only find delete and get for that preference, no set anywhere:
![grafik](https://github.com/user-attachments/assets/dbf60333-552e-4eb3-933e-e0b4884f83cb)

Reply:
@st3iny any ideas?